mlrun/runtimes/nuclio/application/application.py

```diff
@@ -290,6 +337,89 @@ def with_source_archive(
             target_dir=target_dir,
         )
 
+    @classmethod
+    def get_filename_and_handler(cls) -> (str, str):
+        reverse_proxy_file_path = pathlib.Path(__file__).parent / "reverse_proxy.go"
+        return str(reverse_proxy_file_path), "Handler"
+
+    def create_api_gateway(
+        self,
+        path: str = None,
+        ports: list[int] = None,
+        authentication_mode: APIGatewayAuthenticationMode = None,
+        authentication_creds: tuple[str] = None,
+    ):
+        api_gateway = APIGateway(
+            APIGatewayMetadata(
+                name=self.metadata.name,
+                namespace=self.metadata.namespace,
+                labels=self.metadata.labels,
+                annotations=self.metadata.annotations,
+            ),
+            APIGatewaySpec(
+                functions=[self],
+                project=self.metadata.project,
+                path=path,
+                ports=mlrun.utils.helpers.as_list(ports) if ports else None,
+            ),
+        )
+
+        authentication_mode = (
+            authentication_mode
+            or mlrun.mlconf.function.application.default_authentication_mode
+        )
+        if authentication_mode == APIGatewayAuthenticationMode.ACCESS_KEY:
+            api_gateway.with_access_key_auth()
+        elif authentication_mode == APIGatewayAuthenticationMode.BASIC:
+            api_gateway.with_basic_auth(*authentication_creds)
+
+        db = mlrun.get_run_db()
+        api_gateway_scheme = db.store_api_gateway(
+            api_gateway=api_gateway.to_scheme(), project=self.metadata.project
+        )
+        self.status.api_gateway_name = api_gateway.metadata.name
+        self.status.api_gateway = APIGateway.from_scheme(api_gateway_scheme)
+        self.status.api_gateway.wait_for_readiness()
+
+    def invoke(
+        self,
+        path: str,
+        body: typing.Union[str, bytes, dict] = None,
+        method: str = None,
+        headers: dict = None,
+        dashboard: str = "",
+        force_external_address: bool = False,
+        auth_info: AuthInfo = None,
+        mock: bool = None,
```

Comment:
`auth_info` should be initialised and passed by user, right? Why not aligning this logic with how it is done in API Gateway class which means just passing the credentials.

Reply:
Because `invoke` is inherited from the base class `RemoteRuntime`.
Changing it will break backwards compatibility.